File: 217-contains-duplicate/contains-duplicate.java
class Solution {
    public boolean containsDuplicate(int[] nums) {
        /* using HashMap
        HashMap<Integer,Integer> map = new HashMap<>();
        for(int num : nums){
            if(map.containsKey(num)){
                map.put(num,map.getOrDefault(num,0)+1);
            }
            else{
                map.put(num,1);
            }
        }
        for(Integer key : map.keySet()){
            if(map.get(key)>=2){
                return true;
            }
        }
        return false;
        */
        //Using HashSet
        HashSet<Integer> set = new HashSet<>();
        for(int num :  nums){
            if(set.contains(num)){
                return true;
            }
            set.add(num);
        }
        return false;
    }
}

File: 217-contains-duplicate/contains-duplicate.c
int compare(const void *a,const void *b){
    return (*(int*)a-*(int*)b);
}
bool containsDuplicate(int* nums, int n) {
    if(n<=1||nums==NULL){
        return 0;
    }
    qsort(nums,n,sizeof(int),compare);
    for(int i=1;i<n;i++){
        if(nums[i]==nums[i-1]){
            return 1;
        }
    }
    return 0;
}